playbooks/roles/filebeat/templates/filebeat.yml.j2

```diff
@@ -0,0 +1,10 @@
+filebeat:
+  registry_file: "{{ filebeat_registry_file }}"
+
+  prospectors:
+    {% for prospector in filebeat_prospectors %}
+    - {{ filebeat_common_prospector | combine(prospector, recursive=True) }}
+    {% endfor %}
+
+output:
+  logstash: {{ filebeat_logstash }}
```

Comment:
This file ends up looking kinda ugly, but I tried a lot of things to make it prettier. If anyone has ideas of how to make it clean while keeping it as valid YML and persisting the for loop above, do let me know.

Reply:
Since this is a YAML file, I wouldn't define a template for it.  Just store the dictionary in an Ansible variable, and write it to a file using the `to_nice_yaml` filter.  This should render nicer YAML than you get with the current version.

Reply:
The issue is that above I loop through the prospectors and merge them with the common prospector. I don't know if I can achieve something similar, of merging a dictionary with all dictionaries in a list, without a for loop, in plain YAML. Is there a way?

Reply:
Ansible's "type system" (in quotes, since it doesn't really deserve the name) is rather arcane, so I can't foresee whether the result of the Jinja2 loop will be implicitly converted back to a YAML dictionary, or whether it's possible to explicitly convert it back.  I'm rather confident that this would be possible, e.g. using the `from_yaml` Jinja2 filter, but I don't care too much about the rendered template being ugly – your code looks fine, and this matters a lot more, so let's just not spend more time on this.

Reply:
:+1: